src/ui/features/xp-drop/XpDropOnboarding/XpDropOnboarding.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useRef } from 'react';
+import type { To } from 'react-router-dom';
+import { Link, Route, Routes } from 'react-router-dom';
+import { PageColumn } from 'src/ui/components/PageColumn';
+import { NavigationTitle } from 'src/ui/components/NavigationTitle';
+import { VStack } from 'src/ui/ui-kit/VStack';
+import { UIText } from 'src/ui/ui-kit/UIText';
+import { PageStickyFooter } from 'src/ui/components/PageStickyFooter';
+import { Button } from 'src/ui/ui-kit/Button';
+import { PageBottom } from 'src/ui/components/PageBottom';
+import type { HTMLDialogElementInterface } from 'src/ui/ui-kit/ModalDialogs/HTMLDialogElementInterface';
+import { BottomSheetDialog } from 'src/ui/ui-kit/ModalDialogs/BottomSheetDialog';
+import LevelsSrc from '../assets/levels.png';
+import Levels2xSrc from '../assets/levels@2x.png';
+import NewHomeForDnaSrc from '../assets/new-home-for-dna.png';
+import NewHomeForDna2xSrc from '../assets/new-home-for-dna@2x.png';
+import QuestsSrc from '../assets/quests.png';
+import Quests2xSrc from '../assets/quests@2x.png';
+import Rewards2xSrc from '../assets/rewards@2x.png';
+import RewardsSrc from '../assets/rewards.png';
+import { ExitConfirmationDialog } from '../components/ExitConfirmationDialog';
+import { XpDropScoring } from './XpDropScoring';
+
+import * as styles from './styles.module.css';
+
+function WelcomeTo({ title }: { title: string }) {
+  return (
+    <VStack gap={0}>
+      <UIText kind="headline/h3" color="var(--neutral-800)">
+        Welcome to
+      </UIText>
+      <UIText kind="headline/hero" className={styles.gradientText}>
+        {title}
+      </UIText>
+    </VStack>
+  );
+}
+
+function OnboardingStep({
+  title,
+  text,
+  imageSrc,
+  image2xSrc,
+  nextLocation,
+  buttonText,
+}: {
+  title: string;
+  text: string;
+  imageSrc: string;
+  image2xSrc: string;
+  nextLocation: To;
+  buttonText: string;
+}) {
+  // TODO: Show confirmation dialog when the user attempts to navigate back
```

Comment:
Should we finish this before merge?

Reply:
Not really, that requires us to update the react router package, we agreed to do it later, but I decided to leave the TODO comments here to make sure we won't forget